internal/testdata/options/test_editions.proto

```diff
@@ -0,0 +1,80 @@
+edition = "2023";
+
+import "google/protobuf/any.proto";
+import "google/protobuf/descriptor.proto";
+
+message Foo {
+  string name = 1;
+  Foo foo = 2 [
+    (any) = {
+      [type.googleapis.com/Foo]: {
+        Foo: {
+          name: "abc",
+          Foo: { name: "xyz" }
+        }
+      }
+    },
+    features.message_encoding = DELIMITED
+  ];
```

Comment:
This is the dependency cycle mention in the PR description: we have to interpret the options for this field (to see this `feature.message_encoding` option) before we can know how to interpret the other options for the field, since the above option refers to this very field in the message literal.